### PR TITLE
Improved detector parameter consistency.

### LINKED ISSUE #703: Improve detector parameter consistency
Many detectors require specific parameters that are specific to their detection approach. However, there are some general parameters that are shared by many detectors and I think make sense to standardize for all detectors. After a quick review I think that these are the following parameters:

- id
- type
- persistence_id
- suppress
- output_event_handlers
- ignore_list
- constraint_list
- output_logline
- learn_mode

Looking at the AnalysisValidationSchema, many detectors only have parts of these parameters implemented, e.g., the enhanced combo detector does not have constraint and ignore lists. Note that I would only do this for all detector components (that end with <...>Detector.py) and not for all analysis components, since they do not need all of these parameters; however, we could also find other sets of parameters that should be standardized for these groups, for example, for all <...>Rules.py. 

This would improve the overview in the aminer docs, since it would not be necessary to copy the same parameters over and over for each component.

### LINKED ISSUE #1146: DeepSourceIssue: Consider using f-strings (PYL-C0209)
DeepSource has stated the new issue "Consider using f-strings (PYL-C0209)"
Here the commonly used format strings ```'String %s' % string``` should be replaced by ```f"String {string}"```.
Please standardize the formate for all files.

### PR DESCRIPTION
**Pull request by ernstleierzopf**
# Make sure these boxes are signed before submitting your Pull Request -- thank you.

# Must haves
- [x] I have read and followed the contributing guide lines at https://github.com/ait-aecid/logdata-anomaly-miner/wiki/Git-development-workflow
- [x] Issues exist for this PR
- [x] I added related issues using the "Fixes #<issue-id>"-notations
- [x] This Pull-Requests merges into the "development"-branch

Fixes #703
Fixes #1146 

# Submission specific

- [ ] This PR introduces breaking changes
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

# Describe changes:
Refactored multiple variable names and implemented a default __init__ method in the AtomHandlerInterface.

The AtomHandlerInterface.__init__() method now contains (unfinished yet) all possible arguments for the analysis components. This ensures, that duplicate variable names are removed/renamed. Also type checks of the arguements may be done in future in the superclass. The method sets all non-None attributes to the self object. To ensure that no "defined outside __init__()" issues are found, some attributes still need to be set to None.

added mutable_default_args in the AtomHandlerInterface.__init__ method.

mutable_default_args is a list of arguments which contain mutable values and should be set to their default value indicated by the argument name (i.e. ending with list or dict).

Changed variable names: 
auto_include_flag -> learn_mode
output_log_line -> output_logline
path_list, paths, match_value_path_list, value_path_list -> target_path_list
path, property_path -> target_path
output_event_handlers -> anomaly_event_handlers
value_list -> target_value_list
sorted_log_lines -> sorted_loglines

Removed max_artifacts_a_for_single_b in TimeCorrelationViolationDetector as it is not used by the detector.
Removed unused get_id and get_child_elements methods in parsing components.

### CODE REVIEW

File: changelog [hunk 1]
```diff
@@ -4,7 +4,7 @@ logdata-anomaly-miner (2.5.1) unstable; urgency=low
   * Fixed index out of range if matches are empty in JsonModelElement array.
   * EFD: Fixed problem that appears with empty windows
   * EFD: Enabled immediate detection without training, if both limits are set
-  * EFD: Fixed bug related to auto_include_flag
+  * EFD: Fixed bug related to learn_mode
```
Comment (landauermax) [inline changelog:7]: Please undo, changelog must not be changed

File: changelog [hunk 2]
```diff
@@ -227,7 +227,7 @@ logdata-anomaly-miner (2.1.0) unstable; urgency=low
   * Extended demo-config
   * Renamed whitelist to allowlist
   * Added warnings for non-existent resources
-  * Changed default of auto_include_flag to false
+  * Changed default of learn_mode to false
```
Comment (landauermax) [inline changelog:230]: Please undo, changelog must not be changed

File: changelog [hunk 3]
```diff
@@ -431,7 +431,7 @@ logdata-anomaly-miner (0.0.5) xenial; urgency=low
     from aminer.input import SimpleByteStreamLineAtomizerFactory
     analysis_context.atomizer_factory=SimpleByteStreamLineAtomizerFactory(
         parsing_model, parsed_atom_handlers, unparsed_atom_handlers,
-        anomaly_event_handlers, default_timestamp_paths=['/model/syslog/time'])
+        anomaly_event_handlers, default_timestamp_path_list=['/model/syslog/time'])
```
Comment (landauermax) [inline changelog:434]: Please undo, changelog must not be changed

File: changelog (hunk 1 above)
Comment (ernstleierzopf) [reply changelog:7]: Done.

File: changelog (hunk 2 above)
Comment (ernstleierzopf) [reply changelog:230]: Done.

File: changelog (hunk 3 above)
Comment (ernstleierzopf) [reply changelog:434]: Done.

**Review by landauermax — APPROVED** [review]
Looks good to me

**Pull request merged**